TribalWars/src/ApplicationCore/Services/TribeService.cs

```diff
@@ -44,7 +47,28 @@ public async Task<ServiceResult> CreateTribe(Tribe tribe, Guid playerId)
 
         public async Task<bool> TribeExists(string tribeName)
         {
-            return await _repository.GetTribeByName(tribeName) != null;
+            return await _tribeRepository.GetTribeByName(tribeName) != null;
+        }
+
+        public async Task<(ServiceResult, TribeDetailsVM)> TribeDetails(Guid playerId)
+        {
+            var tribe = await _tribeRepository.GetTribeByUser(playerId);
+            if (tribe != null)
+            {
+                var tribeUsers = await _tribeUserRepository.GetTribeUsersById(tribe.Id);
+                var owner = tribeUsers
+                    .Find(player => player.TribeRole == Enums.TribeRole.Owner);
+                var details = new TribeDetailsVM
+                {
+                    TribeName = tribe.Name,
+                    Description = tribe.Description,
+                    OwnerName = owner.Player.Nickname,
+                    NumberOfMembers = tribeUsers.Count
+                };
+
+                return (ServiceResult.Success(), details);
+            }
+            return (ServiceResult.Failure(_localizer["TribeDetailsError"]), null);
```

Comment:
That's why making generic ServiceResult<T> would be helpful. In future, reading additional data from tuples on frontend will be pain. 
Also it will give the client a solid structure. Imagine now sending (details, ServiceResult) - reading from Item1 will give different result than everywhere else